packages/camera/camera_android_camerax/android/src/main/java/io/flutter/plugins/camerax/LiveDataHostApiImpl.java

```diff
@@ -0,0 +1,79 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugins.camerax;
+
+import androidx.annotation.NonNull;
+import androidx.lifecycle.LifecycleOwner;
+import androidx.lifecycle.LiveData;
+import io.flutter.plugin.common.BinaryMessenger;
+import io.flutter.plugins.camerax.GeneratedCameraXLibrary.LiveDataHostApi;
+import java.util.Objects;
+
+/**
+ * Host API implementation for {@link LiveData}.
+ *
+ * <p>This class may handle instantiating and adding native object instances that are attached to a
+ * Dart instance or handle method calls on the associated native class or an instance of the class.
+ */
+public class LiveDataHostApiImpl implements LiveDataHostApi {
+  private final BinaryMessenger binaryMessenger;
+  private final InstanceManager instanceManager;
+  private LifecycleOwner lifecycleOwner;
+
+  /**
+   * Constructs a {@link LiveDataHostApiImpl}.
+   *
+   * @param binaryMessenger used to communicate with Dart over asynchronous messages
+   * @param instanceManager maintains instances stored to communicate with attached Dart objects
+   */
+  public LiveDataHostApiImpl(
+      @NonNull BinaryMessenger binaryMessenger, @NonNull InstanceManager instanceManager) {
+    this.binaryMessenger = binaryMessenger;
+    this.instanceManager = instanceManager;
+  }
+
+  /** Sets {@link LifecycleOwner} used to observe the camera state if so requested. */
+  public void setLifecycleOwner(LifecycleOwner lifecycleOwner) {
+    this.lifecycleOwner = lifecycleOwner;
+  }
+
+  /**
+   * Adds an {@link Observer} with the specified identifier to the observers list of this instance
+   * within the lifespan of the {@link lifecycleOwner}.
+   */
+  @Override
+  @SuppressWarnings("unchecked")
```

Comment:
It looks like this file is responsible for the creation of the unchecked warning. Is it not possible to modify getLiveDataInstance to avoid this warning.

Reply:
This was generated, but wouldn't we have to know the type of the instance passed to the method to avoid the warning?